analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/LocksReleasedAllPaths.Monitor.Conditions.cs

```diff
@@ -326,5 +326,30 @@ public int MyProperty
                 }
             }
         }
+
+        public void Lambda(bool condition)
+        {
+            Action action = () =>
+            {
+                Monitor.Enter(obj); // // FN, lambdas are not yet supported
```

Comment:
I do not agree, it should work for both lambdas (parenthesized and nonparenthesized) and anonymous functions, see this:
https://github.com/SonarSource/sonar-dotnet/blob/88af512b7bba6f3095c1d42f4b39d2b422d2799d/analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/SymbolicExecutionRunnerTest.cs#L214-L235

Please add those two missing syntaxes as well, as these walkers will be the inspiration for other rules as well in the future.